pkg/authorization/authorizer/accessrestriction/accessrestriction.go

```diff
@@ -0,0 +1,175 @@
+package accessrestriction
+
+import (
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apiserver/pkg/authentication/user"
+	"k8s.io/apiserver/pkg/authorization/authorizer"
+	"k8s.io/kubernetes/plugin/pkg/auth/authorizer/rbac"
+
+	userapiv1 "github.com/openshift/api/user/v1"
+	userlisterv1 "github.com/openshift/client-go/user/listers/user/v1"
+	"github.com/openshift/origin/pkg/authorization/apis/authorization"
+	authorizationlister "github.com/openshift/origin/pkg/authorization/generated/listers/authorization/internalversion"
+
+	"github.com/golang/glog"
+)
+
+func NewAuthorizer(accessRestrictionLister authorizationlister.AccessRestrictionLister, userLister userlisterv1.UserLister, groupLister userlisterv1.GroupLister) authorizer.Authorizer {
+	return &accessRestrictionAuthorizer{
+		accessRestrictionLister: accessRestrictionLister,
+		userLister:              userLister,
+		groupLister:             groupLister,
+	}
+}
+
+type accessRestrictionAuthorizer struct {
+	accessRestrictionLister authorizationlister.AccessRestrictionLister
+	userLister              userlisterv1.UserLister
+	groupLister             userlisterv1.GroupLister
+}
+
+func (a *accessRestrictionAuthorizer) Authorize(requestAttributes authorizer.Attributes) (authorizer.Decision, string, error) {
+	accessRestrictions, err := a.accessRestrictionLister.List(labels.Everything())
+	if err != nil {
+		// fail closed (but this should never happen because it means some static generated code is broken)
+		return authorizer.DecisionDeny, "cannot determine access restrictions", err
+	}
+
+	// check all access restrictions and only short circuit on affirmative deny
+	for _, accessRestriction := range accessRestrictions {
```

Comment:
it just now occurs to me that this makes some fun cycles....

Reply:
I expect online to have exactly one access restriction based on what they restrict today.

Reply:
@enj what makes you think this will stay restricted to online? If it is unique to online, it should be a separate component optionally installed by online.